OSDOCS-2567: Overview of images

Amrita42 opened the pull request:
Applies to 4.6+

[Jira](https://issues.redhat.com/browse/OSDOCS-2567)

[Preview](https://deploy-preview-38914--osdocs.netlify.app/openshift-enterprise/latest/openshift_images)

@bmcelvee please do an initial peer review.

**Peer review squad**  a query >> Also the current structure first talks about 
**Configuring the Cluster Samples Operator** But imo I think we should put the module **Understanding containers, images, and image streams** first. Please let me know your thoughts.  

@xiuwang please verify the technical veracity of the content.

xiuwang commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,60 @@
+** Image streams
+
+[discrete]
+== How you can use the Cluster Samplers Operator
```
s/Samplers/Samples

xiuwang commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,60 @@
+During the initial start-up, the operator creates the default samples resource to initiate the creation of the image streams and templates.
+You can use the Cluster Samples Operator to manage the sample image streams and templates stored in the `openshift` namespace.
+
+As a cluster administrator, you can use the Cluster Samplers Operator to:
```
s/Samplers/Samples

xiuwang commented:
/lgtm

bmcelvee commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,60 @@
+[discrete]
+== How you can use the Cluster Samples Operator
+
+During the initial start-up, the operator creates the default samples resource to initiate the creation of the image streams and templates.
```
Suggestion based on IBM style (it may be written as start-up in the related assembly, so this is optional):
s/start-up/startup _or_ s/start-up/installation

Amrita42 replied:
https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s >> sounds good I would opt for `startup` and replace existing four instances of `start-up` with `startup`

bmcelvee commented:
Just one optional suggestion. Otherwise, LGTM! Thank you!

Amrita42 commented:
@bmcelvee Thanks! Whats your thought on >> Also the current structure first talks about
**Configuring the Cluster Samples Operator** But imo I think we should put the module **Understanding containers, images, and image streams** first. Please let me know your thoughts.

jeana-redhat reviewed (commented):
You asked in your comment about moving "Understanding containers, images, and image streams" ahead of "Configuring the Cluster Samples Operator" and "Using the Cluster Samples Operator with an alternate registry" in the organization of this book. I think that's a great idea :) 

What do you think about going a step further and incorporating "Understanding containers, images, and image streams" into the new overview assembly? To me, the existing assembly already felt like an overview, it just wasn't called that or placed up front like and overview ought to be.

Amrita42 commented:
@jeana-redhat  Incorporated `Understanding containers, images, and image streams` in the new overview assembly.  Removed unnecessary words, corrected instances of passive tense, etc.   But I could have missed something, do review :)

jeana-redhat commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,127 @@
+==== Image stream triggers
+An image stream trigger causes a specific action when an image stream tag changes. For example, importing can cause the value of the tag to change, which causes a trigger to fire when there are deployments, builds, or other resources listening for those.
+
+[discrete]
```
ID here

jeana-redhat commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,127 @@
+** xref:../openshift_images/samples-operator-alt-registry.adoc#samples-operator-alt-registry[Use the operator with an alternate registry].
+
+
+[discrete]
```
ID here

jeana-redhat commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,127 @@
+A template is a definition of an object to be replicated.
+You can use xref:../openshift_images/using-templates.adoc#using-templates[templates] to build and deploy configurations.
+
+[discrete]
```
ID here

jeana-redhat commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,127 @@
+
+toc::[]
+
+[discrete]
```
Unless the overview assemblies are not supposed to have IDs, this should be a unique ID.

jeana-redhat reviewed (commented):
This is looking good. I think you might have done a bit of extra work here, since most of these items can be used from their existing modules. I've left some notes on how to weave those into the supplementary content and framing that you've added.

Amrita42 commented on openshift_images/index.adoc:
```diff
@@ -0,0 +1,127 @@
+The image stream metadata is stored in the etcd instance along with other cluster information.
+
+[discrete]
+==== Why use image streams
```
I have added to the heading of the module itself.

Amrita42 commented:
@jeana-redhat please review

jeana-redhat reviewed (commented):
LGTM

jeana-redhat commented:
/cherrypick enterprise-4.10

jeana-redhat commented:
/cherrypick enterprise-4.9

jeana-redhat commented:
/cherrypick enterprise-4.8

jeana-redhat commented:
/cherrypick enterprise-4.7

jeana-redhat commented:
/cherrypick enterprise-4.6

Amrita42 commented:
Adding related PRs for 4.6 and 4.7 : https://github.com/openshift/openshift-docs/pull/39855  and https://github.com/openshift/openshift-docs/pull/39863
